[6.5][.NET] Added section: Exchange Data Between Applications (#1500)
Samsung/tizen-docs: docs/application/dotnet/guides/exchange-data-between-apps/overview.md

# Exchange data between applications

> With increasing complexity of the project, it is important to separate an application functionality into smaller, independent components. The advantage of this is:

Reviewer: ```suggestion
With increasing complexity of the project, it is important to separate an application functionality into smaller and independent components. The advantages of this are:
```
```suggestion
With increasing complexity of the project, it is important to separate an application functionality into smaller, independent components. The advantage of this is:
```

![App Control](./media/overview_app_ctrl.gif)

## Data control

Data control allows you to read and modify data stored and provided by another application, and monitor changes in that data. The application storing and controlling the data is called as DataControl provider application. The application using the data is called as DataControl consumer application. A single DataControl provider can serve multiple DataControl consumers.

![Data Control](./media/overview_data_ctrl.png)

## Events broadcast and subscription

> Event is a broadcast message sent by application and all applications which want to listen receive this event. A set of specific platform events provided by the Tizen operating system are as follows: 

Reviewer: Please check if following suggestion looks fine? If yes, kindly update accordingly.

```suggestion
An event is a broadcast message delivered by the application to all other applications that want to listen. A set of specific platform events provided by the Tizen operating system are as follows: 
```

Author: Done

| `SystemEvents.UsbStatus.EventName`         | When the USB state is `SystemEvents.UsbStatus.StatusValueConnected`.     |
| `SystemEvents.EarjackStatus.EventName`     | When the earjack state is `SystemEvents.EarjackStatus.StatusValueConnected`. |
| `SystemEvents.IncomingMsg.EventName`       | When the `SystemEvents.IncomingMsg.TypeKey` and `SystemEvents.IncomingMsg.IdKey` exist.  |
| `SystemEvents.WifiState.EventName`         | When the Wi-Fi state is `SystemEvents.WifiState.StateValueConnected`.   |

Events used to launch app are called as `launch events`.

![Event Broadcast](./media/overview_event.png)

## Message port

Message port provides encrypted communication using author certificate. If data exchanged between two applications should be secured, you could use Message Port API. Secure connection could be established only for applications using the same certificate. The following picture shows possible trusted data exchange between two apps. Possible connection is marked using green color. Communication between apps using `A certificate` and `B certificate` is not allowed.

![Message Port](./media/overview_msgport.png)

## Related information
* Dependencies

>   -   Tizen 4.0 and Higher

Author: This PR was raised to 6.5 branch to avoid conflicts. Regarding to dependency tizen 4.0 and higher is ok.
